frame/system/src/lib.rs

```diff
@@ -951,12 +949,27 @@ impl<T: Trait> Module<T> {
 		<ExtrinsicsRoot<T>>::put(txs_root);
 
 		if let InitKind::Full = kind {
+			sp_io::storage::accumulator_commit(&Events::<T>::storage_value_final_key()[..]);
 			<Events<T>>::kill();
 			EventCount::kill();
 			<EventTopics<T>>::remove_all();
 		}
 	}
 
+	/// Update pending storage changes, if any.
+	pub fn update_accumulators() {
```

Comment:
Why is this public?

Reply:
Maybe superfluous now, let me check if all works when wired up within the `system` (now this method is called in `executive` to flush changes between transactions)